internal/db/update/schema.go

```diff
@@ -85,6 +85,13 @@ func (s *SchemaUpdate) Ensure(db *sql.DB) (int, error) {
 				return err
 			}
 
+			// updateFromV2 introduces the notion of internal and external APi extensions. It needs to be run before we check the API extensions version
+			// in the check hook.
+			err = updateFromV2(ctx, tx)
+			if err != nil {
+				return err
+			}
```

Comment:
I'm uneasy about this update being another special case, I thought the idea was that after #94 we'll have a slice of update functions to run, rather than specific ones called here. 

The check hook ought to fail if the schema versions are different before it checks for differences in API version. Then we will know that the update has been applied on all members. 

Also it looks like there's a misspelling of `API` here (lowercase `i`).

Reply:
So the schema/API update process looks like this, which might help describe the problem a bit:

* Open the database
* Fetch the schema and API versions of all cluster members to compare against our own.
* Update the row corresponding to ourselves in `internal_cluster_members` with the API and schema versions determined when the daemon started.
* Compare the updated schema/API versions to the rest of the cluster
* Potentially error out if we are behind
* By this point we are sure our daemon is valid, so actually run the schema updates that we are missing
* Wait for other nodes if their version is behind


We need the `internal_cluster_members` table to actually have all that information. So we're stuck in a loop where we need `updateFromV2` to have run before we check for any schema updates. 

That leaves us with two rather ugly possible solutions. Either we duplicate the whole block, to fully separate API and schema version checking, which means we need to have unique queries for schema and API updates and verification. Or instead we can manually call `updateFromV2` before we attempt to fetch from the `internal_cluster_members` table for the first time.

Reply:
I'm not clear on why the whole block needs to be duplicated to perform the API version check though.

Once we've passed the schema check, can we not query for each nodes API extensions and just check that the contents are identical?

Reply:
It's definitely more resilient in case we ever need to make a schema update to the `internal_cluster_members` table again.

Reply:
Yeah that's my worry. We have an established system now for schema updates and while it's a little repetitive to perform some of the same steps for the API extensions it's probably better to do this check separately IMO

Reply:
fixed